src/target/riscv/riscv-013.c

```diff
@@ -2817,6 +2816,8 @@ static int init_target(struct command_context *cmd_ctx,
 	generic_info->read_memory = read_memory;
 	generic_info->data_bits = &riscv013_data_bits;
 	generic_info->print_info = &riscv013_print_info;
+	generic_info->get_impebreak = &riscv013_get_impebreak;
+	generic_info->get_progbufsize = &riscv013_get_progbufsize;
```

Comment:
(Just a side note, no change requested)

I would like to get rid of all these callbacks; hopefully I'll find the time to do that soon.

These callbacks are needed now due to improper division of the code. Most notably, some pieces of code that are specific to RISC-V 0.13 still reside in riscv.c. Once the code is properly relocated and cleaned up, these function pointers / callbacks should become redundant, and direct function calls will be possible.

Reply:
Wouldn't it be confusing? And how to guarantee that function called for the appropriate type? Just assume that this is always correct or use assert?

Reply:
Most of the RISC-V code is specific to riscv-0.11 vs. riscv-0.13 debug spec versions. For that reason the code for 0.11 should directly call the 0.11 functions, and the same is true for 0.13.

To use `get_impebreak` and `get_progbufsize` as examples:

These callbacks are specific for riscv-0.13. For that reason:
- The code for 0.11 should never use any of them.
- The code for 0.13 should call directly `riscv013_get_impebreak()` and `riscv013_get_progbufsize()` (once the code is properly re-structured into functions and modules).

I'm not sure if I managed to explain my point clearly. Anyway, I hope to be able to make a PR with the proposed changes soon.

Reply:
OK, thank you for the explanation!